anesthetic/utils.py

```diff
@@ -378,15 +414,15 @@ def sample_compression_1d(x, w=None, ncompress=True):
     if ncompress is False:
         return x, w
     elif ncompress is True:
-        ncompress = channel_capacity(w)
+        ncompress = effective_samples(w)
```

Comment:
We should propagate the `gamma` kwarg to the various functions and methods that call `effective_samples`.